src/doc/src/reference/build-scripts.md

```diff
@@ -99,41 +99,41 @@ configuration). The stderr output is also saved in that same directory.
 one detailed below.
 
-* [`cargo:rerun-if-changed=PATH`](#rerun-if-changed) --- Tells Cargo when to
+* [`cargo::rerun-if-changed=PATH`](#rerun-if-changed) --- Tells Cargo when to
   re-run the script.
-* [`cargo:rerun-if-env-changed=VAR`](#rerun-if-env-changed) --- Tells Cargo when
+* [`cargo::rerun-if-env-changed=VAR`](#rerun-if-env-changed) --- Tells Cargo when
   to re-run the script.
-* [`cargo:rustc-link-arg=FLAG`](#rustc-link-arg) --- Passes custom flags to a
+* [`cargo::rustc-link-arg=FLAG`](#rustc-link-arg) --- Passes custom flags to a
   linker for benchmarks, binaries, `cdylib` crates, examples, and tests.
-* [`cargo:rustc-link-arg-bin=BIN=FLAG`](#rustc-link-arg-bin) --- Passes custom
+* [`cargo::rustc-link-arg-bin=BIN=FLAG`](#rustc-link-arg-bin) --- Passes custom
   flags to a linker for the binary `BIN`.
-* [`cargo:rustc-link-arg-bins=FLAG`](#rustc-link-arg-bins) --- Passes custom
+* [`cargo::rustc-link-arg-bins=FLAG`](#rustc-link-arg-bins) --- Passes custom
   flags to a linker for binaries.
-* [`cargo:rustc-link-arg-tests=FLAG`](#rustc-link-arg-tests) --- Passes custom
+* [`cargo::rustc-link-arg-tests=FLAG`](#rustc-link-arg-tests) --- Passes custom
   flags to a linker for tests.
-* [`cargo:rustc-link-arg-examples=FLAG`](#rustc-link-arg-examples) --- Passes custom
+* [`cargo::rustc-link-arg-examples=FLAG`](#rustc-link-arg-examples) --- Passes custom
   flags to a linker for examples.
-* [`cargo:rustc-link-arg-benches=FLAG`](#rustc-link-arg-benches) --- Passes custom
+* [`cargo::rustc-link-arg-benches=FLAG`](#rustc-link-arg-benches) --- Passes custom
   flags to a linker for benchmarks.
-* [`cargo:rustc-link-lib=LIB`](#rustc-link-lib) --- Adds a library to
+* [`cargo::rustc-link-lib=LIB`](#rustc-link-lib) --- Adds a library to
   link.
-* [`cargo:rustc-link-search=[KIND=]PATH`](#rustc-link-search) --- Adds to the
+* [`cargo::rustc-link-search=[KIND=]PATH`](#rustc-link-search) --- Adds to the
   library search path.
-* [`cargo:rustc-flags=FLAGS`](#rustc-flags) --- Passes certain flags to the
+* [`cargo::rustc-flags=FLAGS`](#rustc-flags) --- Passes certain flags to the
   compiler.
-* [`cargo:rustc-cfg=KEY[="VALUE"]`](#rustc-cfg) --- Enables compile-time `cfg`
+* [`cargo::rustc-cfg=KEY[="VALUE"]`](#rustc-cfg) --- Enables compile-time `cfg`
   settings.
-* [`cargo:rustc-env=VAR=VALUE`](#rustc-env) --- Sets an environment variable.
-* [`cargo:rustc-cdylib-link-arg=FLAG`](#rustc-cdylib-link-arg) --- Passes custom
+* [`cargo::rustc-env=VAR=VALUE`](#rustc-env) --- Sets an environment variable.
+* [`cargo::rustc-cdylib-link-arg=FLAG`](#rustc-cdylib-link-arg) --- Passes custom
   flags to a linker for cdylib crates.
-* [`cargo:warning=MESSAGE`](#cargo-warning) --- Displays a warning on the
+* [`cargo::warning=MESSAGE`](#cargo-warning) --- Displays a warning on the
   terminal.
-* [`cargo:KEY=VALUE`](#the-links-manifest-key) --- Metadata, used by `links`
+* [`cargo::KEY=VALUE`](#the-links-manifest-key) --- Metadata, used by `links`
```

Comment:
I think something we lost track of in our discussions is changing away from `cargo:KEY=VALUE` because that is what got us into this mess in the first place (we can't add new keys without possibly breaking compatibility).

From [my comment](https://github.com/rust-lang/cargo/issues/11461#issuecomment-1339822618)

> Ideally, cargo:KEY=VALUE would be changed to cargo:metadata=KEY=VALUE (name to be bikesheded) and we can extend as needed.

So `cargo:KEY=VALUE` needs to still be supported not with `::` syntax but instead we support `cargo::metadata=KEY=VALUE`

Reply:
Feel free to make the change with the proposed name (or if you find a better one, that) and I'll drive us getting to consensus on *some* name and get back to you.

Reply:
`cargo:key=value` is the general syntax.  It had a catch all case where if `key` was not known, it would fallback to treating it as a metadata for `link`.  This `links` metadata catch-all was generally written in the docs as `cargo:KEY=VALUE` (notice the case).  My intention was to communicate that the `links` metadata case was to be put under the `metadata` key.

So for the old syntax we'd have:
- `cargo:rustc-flags=VALUE`
- `cargo:rustc-link-lib=VALUE`
- `cargo:KEY=VALUE`

And for the new syntax version we'd have:
- `cargo::rustc-flags=VALUE`
- `cargo::rustc-link-lib=VALUE`
- `cargo::metadata=KEY=VALUE`

This will allow us to add new directives without breaking compatibility because we would not be shadowing a catch-all case.

Reply:
Got it. Thanks for your explanation.